app/models/service_retire_task.rb

```diff
@@ -30,6 +30,9 @@ def after_request_task_create
     Service.where(:id => options[:src_ids]).each do |parent_svc|
       if create_subtasks?(parent_svc)
         _log.info("- creating service subtasks for service task <#{self.class.name}:#{id}>, service <#{parent_svc.id}>")
+        if parent_svc.children.pluck(:name).include?(parent_svc.name)
```

Comment:
At a minimum, I think a better SQL is `parent_svc.children.where(:name => parent_svc.name).exists?`